agent/engine/docker_client.go

```diff
@@ -493,15 +487,9 @@ func (dg *dockerGoClient) CreateContainer(config *docker.Config,
 	case resp := <-response:
 		return resp
 	case <-ctx.Done():
-		// Context has either expired or canceled. If it has timed out,
-		// send back the DockerTimeoutError
-		err := ctx.Err()
-		if err == context.DeadlineExceeded {
-			return DockerContainerMetadata{Error: &DockerTimeoutError{timeout, "created"}}
-		}
-		// Context was canceled even though there was no timeout. Send
-		// back an error.
-		return DockerContainerMetadata{Error: &CannotCreateContainerError{err}}
+		// Context has expired, send back the DockerTimeoutError
```

Comment:
Why was this modified. I think there's a case to be made for the previous code, where we distinguish between cancelled and timed out contexts.

Reply:
As Derek said, when the caller wants to cancel the context, the caller already know the reason,  the logic to handle the timeout or cancel is nothing but report the reason of the error.

Reply:
That's not strictly true. If you look at https://github.com/aws/amazon-ecs-agent/pull/1329/files#diff-b9aa12c3425c34d6f330f28407f779f6R835, you'll see that the caller (owner of the context) is not the entity that's dealing with the error. I think there's value in the way we were previously handling errors until we figure out a better alternative.

Reply:
OK, cool, will change accordingly.